src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +70,812 @@ const ActorInit En_Fz_InitVars = {
+
+        vec1.x = 0.0f;
+        vec1.y = -2.0f;
+        vec1.z = 20.0f;
+        Matrix_MultVec3f(&vec1, &velocity);
+
+        if ((this->counter % 8) == 0) {
+            isTimerMod8 = true;
+        }
+
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, isTimerMod8);
+        pos.x += (velocity.x * 0.5f);
+        pos.y += (velocity.y * 0.5f);
+        pos.z += (velocity.z * 0.5f);
+        EnFz_SetIceSmokeFreezing(this, &pos, &velocity, &accel, 2.0f, 25.0f, primAlpha, false);
+    }
+}
+
+void EnFz_Update(Actor* thisx, GlobalContext* globalCtx) {
+    static EnFzSpawnIceSmokeFunc sSpawnIceSmoke[] = {
+    EnFz_SpawnIceSmokeHiddenState,  // Hiding
+    EnFz_SpawnIceSmokeGrowingState, // Fully grown
+    EnFz_SpawnIceSmokeActiveState,  // Growing or Shrinking to/from hiding
+    EnFz_SpawnIceSmokeActiveState,  // melting from fire attack
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fz/func_80A218A8.s")
+    EnFz* this = THIS;
+    CollisionCheckContext* colChkCtx;
+
+    this->counter++;
+
+    if (this->unusedTimer1) {
+        this->unusedTimer1--;
+    }
+
+    if (this->timer) {
+        this->timer--;
+    }
+
+    if (this->unusedTimer2) {
+        this->unusedTimer2--;
+    }
+
+    Actor_SetHeight(&this->actor, 50.0f);
+    EnFz_ApplyDamage(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+
+    if (this->isDespawning == false) {
+        Collider_CylinderUpdate(&this->actor, &this->collider1);
+        Collider_CylinderUpdate(&this->actor, &this->collider2);
+
+        if (this->isFreezing) {
+            if (this->actor.dmgEffectTimer == 0) {
+                colChkCtx = &globalCtx->colChkCtx;
+                CollisionCheck_SetAC(globalCtx, colChkCtx, &this->collider1);
+                CollisionCheck_SetAC(globalCtx, colChkCtx, &this->collider2);
+            }
+
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider1);
+        }
+    }
+
+    Math_ApproxF(&this->actor.speedXZ, this->xzSpeed, 0.2f);
+    Actor_MoveForward(&this->actor);
+
+    if (this->isAlwaysTrue) {
+        func_8002E4B4(globalCtx, &this->actor, 20.0f, 20.0f, 20.0f, 5);
+    }
+
+    sSpawnIceSmoke[this->state](this);
+    EnFz_MoveIceSmoke(this, globalCtx);
+}
+
+void EnFz_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    static Gfx* sDLists[] = { 0x06001130,   // Body fully intact           (5 or 6 health)
+                              0x060021A0,   // Top right horn chipped off  (from Freezards perspective)   (3 or 4 health)
+                              0x06002CA0 }; // Entire head chipped off     (1 or 2 health)
+    EnFz* this = THIS;
+    u32 frames;
+    s32 index;
+
+    index = (6 - this->actor.colChkInfo.health) >> 1;
+    // 6 - 0
+    // 5 - 0
+    // 4 - 1
+    // 3 - 1
+    // 2 - 2
+    // 1 - 2
+    // 0 - 3
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_fz.c", 1167);
+
+    if (this->actor.colChkInfo.health == 0) {
+        if (1) {}
```

Comment:
```suggestion
    if (1) {}
    
    if (this->actor.colChkInfo.health == 0) {
```
I prefer this outside the `if`, if that still matches.

Reply:
It matches